internal/cloudapi/v2/errors.go

```diff
@@ -144,24 +146,43 @@ func find(code ServiceErrorCode) *serviceError {
 			return &e
 		}
 	}
-	return &serviceError{ErrorServiceErrorNotFound, http.StatusInternalServerError, "Error does not exist"}
+	return &serviceError{ErrorServiceErrorNotFound, http.StatusInternalServerError, "Error does not exist", ""}
 }
 
 // Make an echo compatible error out of a service error
 func HTTPError(code ServiceErrorCode) error {
-	return HTTPErrorWithInternal(code, nil)
+	return HTTPErrorWithInternal(code, nil, "")
 }
 
 // echo.HTTPError has a message interface{} field, which can be used to include the ServiceErrorCode
-func HTTPErrorWithInternal(code ServiceErrorCode, internalErr error) error {
+func HTTPErrorWithInternal(code ServiceErrorCode, internalErr error, details string) error {
 	se := find(code)
+	se.details = details
```

Comment:
This won't actually assign the details to the error, as the service error is thrown away after constructing the echo error below. It needs to make it all the way to the error handler.